cognite/client/utils/_azure_token_generation.py

```diff
@@ -0,0 +1,40 @@
+import os
+
+from cognite.client.exceptions import CogniteAPIKeyError
+from cognite.client.utils import _auxiliary
+
+
+def _azure_environment_vars_set():
+    azure_api_client_id = os.getenv("AZURE_API_CLIENT_ID")
+    azure_api_client_secret = os.getenv("AZURE_API_CLIENT_SECRET")
+    oidc_authority = os.getenv("OIDC_AUTHORITY")
+
+    return azure_api_client_id is not None and azure_api_client_secret is not None and oidc_authority is not None
+
+
+def _generate_access_token():
+    # Microsoft Authentication Library (MSAL) for Python
+    # https://github.com/AzureAD/microsoft-authentication-library-for-python
+    msal = _auxiliary.local_import("msal")
+    azure_api_client_id = os.getenv("AZURE_API_CLIENT_ID")
+    azure_api_client_secret = os.getenv("AZURE_API_CLIENT_SECRET")
+    oidc_authority = os.getenv("OIDC_AUTHORITY")
+    cognite_base_url = os.getenv("COGNITE_BASE_URL", "https://api.cognitedata.com")
+
+    app = msal.ConfidentialClientApplication(
+        azure_api_client_id, client_credential=azure_api_client_secret, authority=oidc_authority
+    )
+    # When using the client-credentials flow with msal the only valid scope is .default
+    scopes = [f"{cognite_base_url}/.default"]
```

Comment:
Does this result in a usable token? I recall needing to set the scope to something along the lines of `$AAD_APP_REGISTRATION_CLIENT_ID/.default` in order for the resulting token to be useable.

Reply:
The `${cognite_base_ul}` isn't guaranteed to work- it's fully possible to use some other scope if so configured in the IDP. 
In general, this should be something like `{resource_id}/.default` or `{api_identifier}/.default` or just `{scope}`